File: src/main/java/me/nepnep/nepbot/message/command/commands/fun/Xkcd.java
package me.nepnep.nepbot.message.command.commands.fun;

import me.nepnep.nepbot.message.command.ICommand;
import me.nepnep.nepbot.message.command.Category;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;

import java.util.List;

public class Xkcd implements ICommand {
    @Override
    public void execute(List<String> args, GuildMessageReceivedEvent event) {
        int comic = (int) (Math.random() * (2421 - 1)) + 1;
        event.getChannel().sendMessage("https://xkcd.com/" + comic).queue();
    }
    @Override
    public String getInvoke() {
        return "xkcd";
    }
    @Override
    public Permission requiredPermission() {
        return null;
    }
    @Override
    public Category getCategory() {
        return Category.FUN;
    }
    @Override
    public String getDescription() {
        return "Returns a random xkcd comic";
    }
}

File: src/main/kotlin/me/nepnep/nepbot/message/command/commands/fun/Xkcd.kt
package me.nepnep.nepbot.message.command.commands.`fun`

import me.nepnep.nepbot.message.command.Category
import me.nepnep.nepbot.message.command.AbstractCommand
import me.nepnep.nepbot.request
import net.dv8tion.jda.api.entities.GuildMessageChannel
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

class Xkcd : AbstractCommand(
    "xkcd",
    Category.FUN,
    "Returns a random xkcd comic or a specific one"
) {
    override fun execute(args: List<String>, event: MessageReceivedEvent, channel: GuildMessageChannel) {
        when (args.size) {
            0 -> {
                event.jda.httpClient.request("https://c.xkcd.com/random/comic/", {
                    channel.sendMessage(it.request.url.toString()).queue()
                }) {
                    channel.sendMessage("Request failed").queue()
                }
            }
            1 -> {
                val url = "https://xkcd.com/" + args[0]

                event.jda.httpClient.request(url, {
                    if (it.code == 200) {
                        channel.sendMessage(url).queue()
                    } else {
                        channel.sendMessage("Comic does not exist").queue()
                    }
                }) {
                    channel.sendMessage("Request failed").queue()
                }
            }
            else -> {
                channel.sendMessage("Invalid usage").queue()
            }
        }
    }
}